sdk/src/main/java/co/decodable/sdk/pipeline/internal/DecodableSecretImpl.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright Decodable, Inc.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package co.decodable.sdk.pipeline.internal;
+
+import co.decodable.sdk.pipeline.DecodableSecret;
+import co.decodable.sdk.pipeline.exception.SecretNotFoundException;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import org.apache.commons.io.FileUtils;
+
+public class DecodableSecretImpl implements DecodableSecret {
+  private static final String SECRET_DIRECTORY = "/opt/flink/opt/secrets/";
+
+  private final SecretMetadata metadata;
+  private final String value;
+
+  public DecodableSecretImpl(String name) {
+    this(name, SECRET_DIRECTORY);
+  }
+
+  DecodableSecretImpl(String name, String secretDirectory) {
+    if (!secretDirectory.endsWith("/")) {
+      secretDirectory = secretDirectory + "/";
+    }
+    var secretFile = new File(secretDirectory + name);
+    var secretMetadataFile = new File(secretDirectory + name + ".metadata");
+    if (!secretFile.exists() || !secretMetadataFile.exists()) {
+      throw new SecretNotFoundException(
+          String.format(
+              "Secret [%s] not found. Please make sure it is included in this pipeline's properties.",
+              name));
+    }
+    try {
+      this.value = FileUtils.readFileToString(secretFile, StandardCharsets.UTF_8);
+      var objectMapper = new ObjectMapper();
```

Comment:
Could be static and thus instantiated only once? It's quite costly (but also thread-safe).

Reply:
Done